serving/docker/partition/sm_neo_quantize.py

```diff
@@ -0,0 +1,115 @@
+#!/usr/bin/env python
+#
+# Copyright 2024 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"). You may not use this file
+# except in compliance with the License. A copy of the License is located at
+#
+# http://aws.amazon.com/apache2.0/
+#
+# or in the "LICENSE.txt" file accompanying this file. This file is distributed on an "AS IS"
+# BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, express or implied. See the License for
+# the specific language governing permissions and limitations under the License.
+
+import sys
+import logging
+import os
+from types import SimpleNamespace
+from typing import Final
+import torch
+
+from sm_neo_utils import (CompilationFatalError, write_error_to_file,
+                          get_neo_env_vars)
+from utils import extract_python_jar
+#from quantization_properties_manager import QuantizationPropertiesManager
+#from quantize import QuantizationService
+from properties_manager import PropertiesManager
+from partition import PartitionService
+
+PYTHON_CACHE_DIR = '/tmp/djlserving/cache'
+
+
+class NeoQuantizationService():
+
+    def __init__(self):
+        self.args: SimpleNamespace = SimpleNamespace()
+        self.properties_manager: PropertiesManager = None
+        self.compiler_flags: dict = None
+
+        env = get_neo_env_vars()
+        self.INPUT_MODEL_DIRECTORY: Final[str] = env[1]
+        self.OUTPUT_MODEL_DIRECTORY: Final[str] = env[2]
+        self.COMPILATION_ERROR_FILE: Final[str] = env[3]
+        self.COMPILER_CACHE_LOCATION: Final[str] = env[4]
+
+    def update_dataset_cache_location(self):
+        logging.info(
+            f"Updating HuggingFace Datasets cache directory to: {self.COMPILER_CACHE_LOCATION}"
+        )
+        os.environ['HF_DATASETS_CACHE'] = self.COMPILER_CACHE_LOCATION
```

Comment:
What if compiler_cache_location is None. Is that okay? Do we validate them somewhere?

Reply:
We validate here: https://github.com/deepjavalibrary/djl-serving/blob/6ec92723030c7fc70021308a09549a410c8ef29b/serving/docker/partition/sm_neo_utils.py#L30